src/mono/mono/mini/interp/transform.c

```diff
@@ -1107,10 +1131,18 @@ get_data_item_index (TransformData *td, void *ptr)
 	index = td->n_data_items;
 	td->data_items [index] = ptr;
 	++td->n_data_items;
+	g_assertf (index < (UINT32_MAX - 1), "Interpreter data item 32-bit wide index 0x%x for method '%s' overflows", index, td->method->name);
```

Comment:
I think it is pointless to assert for 32bit limit. Nothing will ever work on interp for such methods, not just the data items.